saleor/plugins/webhook/tests/subscription_webhooks/test_create_deliveries_for_subscription.py

```diff
@@ -899,6 +900,76 @@ def test_product_created_multiple_events_in_subscription(
     assert deliveries[0].webhook == webhooks[0]
 
 
+def generate_expected_payload_for_voucher(voucher, voucher_global_id):
+    return json.dumps(
+        {
+            "voucher": {
+                "id": voucher_global_id,
+                "name": voucher.name,
+                "code": voucher.code,
+                "usageLimit": voucher.usage_limit,
+            },
+            "meta": None,
+        }
+    )
+
+
+def test_voucher_created(voucher, subscription_voucher_created_webhook):
+    # given
+    webhooks = [subscription_voucher_created_webhook]
+    event_type = WebhookEventAsyncType.VOUCHER_CREATED
+    voucher_id = graphene.Node.to_global_id("Voucher", voucher.id)
+
+    # when
+    deliveries = create_deliveries_for_subscriptions(event_type, voucher, webhooks)
+
+    # then
+    expected_payload = generate_expected_payload_for_voucher(voucher, voucher_id)
+    assert deliveries[0].payload.payload == expected_payload
+    assert len(deliveries) == len(webhooks)
+    assert deliveries[0].webhook == webhooks[0]
+
+
+def test_voucher_updated(voucher, subscription_voucher_updated_webhook):
+    # given
+    webhooks = [subscription_voucher_updated_webhook]
+    event_type = WebhookEventAsyncType.VOUCHER_UPDATED
+    voucher_id = graphene.Node.to_global_id("Voucher", voucher.id)
+
+    # when
+    deliveries = create_deliveries_for_subscriptions(event_type, voucher, webhooks)
+
+    # then
+    expected_payload = generate_expected_payload_for_voucher(voucher, voucher_id)
+    assert deliveries[0].payload.payload == expected_payload
+    assert len(deliveries) == len(webhooks)
+    assert deliveries[0].webhook == webhooks[0]
+
+
+def test_voucher_deleted(voucher, subscription_voucher_deleted_webhook):
+    # given
+    webhooks = [subscription_voucher_deleted_webhook]
+
+    voucher_query = Voucher.objects.filter(pk=voucher.id)
+    voucher_instances = [voucher for voucher in voucher_query]
+    voucher_query.delete()
```

Comment:
Is it needed? voucher object is already in fixture `voucher`

Reply:
@jakubkuc you are right, but we have 2 cases:
1. delete one object -  so we just save this object `ID` before deleting, then we use `delete()` and assign the saved `ID` back to a deleted object and send this object to `webhook plugin`
2. delete multiple objects using `bulk_delete` - so we have `queryset`, we create a list of objects using this query set, then we use `delete()` on the queryset, queryset is empty after `delete()` so we iterate over the list that we created and send objects to `webhook plugin`

I just used the second option to simulate in test but I changed it to the version that you suggested.